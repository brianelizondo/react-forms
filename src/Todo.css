.Todo{
    padding: 5px 0px;
}
.Todo button{
    margin-left: 5px;
}
.Todo-completed{
    text-decoration:line-through;
    color: gray;
}