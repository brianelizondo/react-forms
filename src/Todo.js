import React from "react";
import "./Todo.css"

function Todo({ task, id, showEditForm, completedTaskStatus, completedTask, deleteTask }){
    const handleEdit = () => {
        showEditForm({ id, task });
    }
    const handleCompleted = () => {
        completedTask(id);
    }
    const handleDelete = () => {
        deleteTask(id);
    }

    const completedButtonText = completedTaskStatus == false ? "Mark as completed" : "Mark as not completed";
    const classStyle = completedTaskStatus == false ? "Todo" : "Todo Todo-completed";

    return (
        <div className={classStyle}>
            { task }
            <button onClick={handleEdit}>Edit</button>
            <button onClick={handleCompleted}>{completedButtonText}</button>
            <button onClick={handleDelete}>X</button>
        </div>
    );
}

export default Todo;