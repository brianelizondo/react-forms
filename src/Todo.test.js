import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

// Smoke Tests
it("'Todo' component renders without crashing", function() {
    render(<Todo />);
});

// Snapshot Tests
it("'Todo' matches snapshot", function() {
    const {asFragment} = render(<Todo task="test" id={123} key={123}/>);
    expect(asFragment()).toMatchSnapshot();
});