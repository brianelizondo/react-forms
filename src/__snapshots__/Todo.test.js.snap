// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'Todo' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    test
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
